lib/collectionspace/tools/xml.rb

```diff
@@ -106,7 +272,93 @@ def self.process_array(xml, label, array)
       end
 
       module Helpers
+        
+        def self.apply_transforms(transforms, groups)
+          replace_config = transforms.select{ |k, v| v.keys.include?('replace') }
+          special_config = transforms.select{ |k, v| v.keys.include?('special') }
+          vocab_config = transforms.select{ |k, v| v.keys.include?('vocab') }
+          authority_config = transforms.select{ |k, v| v.keys.include?('authority') }
+          
+          groups.each{ |group|
+            group.each{ |field, value|
+              if replace_config.keys.include?(field)
+                replacements = replace_config[field]['replace']
+                replacements.each{ |r|
+                  case r['type']
+                  when 'plain'
+                    group[field] = value.gsub!(r['find'], r['replace'])
+                  when 'regexp'
+                    group[field] = value.gsub!(Regexp.new(r['find']), r['replace'])
+                  end
+                }
+              end
 
+              if special_config.keys.include?(field)
+                case special_config[field]['special']
+                when 'boolean'
+                  group[field] = Helpers.to_boolean(value)
+                when 'behrensmeyer_translate'
+                  puts "INFO: field #{field}: BT"
+                  group[field] = Helpers.behrensmeyer_translate(value)
+                end
+              end
+              
+              if vocab_config.keys.include?(field)
+                vocab_name = vocab_config[field]['vocab']
+                group[field] = Helpers.get_vocab(vocab_name, value)
+              end
+
+              if authority_config.keys.include?(field)
+                authority_type = authority_config[field]['authority'][0]
+                authority_name = authority_config[field]['authority'][1]
+                group[field] = Helpers.get_authority(authority_type, authority_name, value)
+              end
+            }
+          }
+        end
+
+        def self.to_boolean(value)
```

Comment:
I could not figure out how to test CSRD.to_boolean and it didn't do what I needed to, so I created this to keep making progress. It seems like a lot of intermediate data wrangling stuff not directly to the XML was already in here when I started, and I've added a bunch more. Happy to move it elsewhere if that makes more sense.